### PR TITLE
Remove defunct PRENET45 from TPL Dataflow

### PR DESCRIPTION
**Pull request by stephentoub**
This PRENET45 compilation constant is legacy from when the library was building in CTPs for .NET 4, but it's not been used for a while and the library won't actually build as written with the flag set.

### CODE REVIEW

File: src/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs [hunk 1]
```diff
@@ -2656,12 +2610,8 @@ private Task ProcessItemAsync(TOutput item)
                         if (_tempSendAsyncTaskList != null && _tempSendAsyncTaskList.Count > 0)
                         {
                             // Consolidate all SendAsync tasks into one
-                            var allSendAsyncTasksConsolidated =
-#if PRENET45
-                            Task.Factory.ContinueWhenAll((Task[])_TempSendAsyncTaskList.ToArray(), s_whenAllAction);
-#else
-                            Task.WhenAll(_tempSendAsyncTaskList);
-#endif
+                            var allSendAsyncTasksConsolidated = Task.WhenAll(_tempSendAsyncTaskList);
```
Comment (weshaggard) [inline src/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs]: Could you please explicitly cast this variable and a few others in in this function to Task?

**Comment by weshaggard** [comment]
As I just sent you an email on if you think we will still have a .NET 4.0 version of this library in the nuget package then we should leave these and complete the code to make it work on that target. I expect that the nuget package will be completely produced by the code here once we get that fully working.

**Comment by terrajobst** [comment]
TPL Dataflow never supported .NET 4.0, so removing the preprocessor symbol should be fine.

File: src/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs (hunk 1 above)
Comment (terrajobst) [inline src/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs]: You mean replacing `var` with `Taks`, right? Casting doesn't make sense to me.

**Comment by terrajobst** [comment]
LGTM

File: src/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs (hunk 1 above)
Comment (weshaggard) [inline src/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs]: Yes I meant to say "explicitly type" to Task. It wasn't obvious to me what the type was so I had to dig deeper.

**Comment by stephentoub** [comment]
@weshaggard, I addressed your feedback about not using var, but rather than just doing it for that one case, I went through the code and applied the guideline more rigorously.  The codebase now only uses var in cases where the type is explicitly stated on the right hand side, in particular if the right hand side is a construction, a cast, or an obvious factory.

**Comment by weshaggard** [comment]
Thanks @stephentoub. LGTM

**Pull request merged**